plugin/src/index.js

```diff
@@ -1,22 +1,25 @@
 // import { preflights } from './preflights.js'
```

Comment:
Might seem like a minor thing but I would move this to a separate file named `presetEngine.js` which would hold the current implementation, and have `index.js` file be responsible for exposing it to the outside world. Im not a fan of using `index`-files for actual implementations

```js
// plugin/src/presetEngine.js
function presetEngine(...) {
  ...
}

export default presetEngine;
```

```js
// plugin/src/index.js
import presetEngine from './presetEngine';

export { presetEngine }
```